src/fileio/dataformat.cxx

```diff
@@ -23,7 +23,7 @@ bool DataFormat::openw(const string &name, int mype, bool append) {
   return openw(base + "." + toString(mype) + "." + ext, append);
 }
 
-bool DataFormat::setLocalOrigin(int x, int y, int z, int offset_x, int offset_y, int offset_z) {
+bool DataFormat::setLocalOrigin(int x, int y, int z, int UNUSED(offset_x),
```

Comment:
Ah, that's what that tiny bell ringing was! I am still none the wiser. I think @johnomotani wrote that code -- John, can you weigh in?

Reply:
I think I did add them when I was trying to get parallel HDF5 I/O working. As far as I remember I thought it ought to work, but didn't have a working MPI-enabled HDF5 to try it out with. I thought at the time that it needed a parallel file-system and I didn't manage to get HDF5 set up on ARCHER with MPI. So offset_x, offset_y, offset_z are implemented and used only in the HDF5 version, so it should be fine to set them as UNUSED here.
[I don't know about the parallel pnetcdf version. I vaguely remember thinking that it should need similar logic to the parallel HDF5 and so need those offsets, but I haven't looked at pnetcdf closely enough to understand what it was trying to do]
The offset_* arguments are meant to be used when writing in parallel because then we don't want to write the guard cells, so the offset tells H5Format where in the local array in memory to write from, so that H5Format doesn't need to know about the mesh. (If I remember right how it was meant to work.)

Anyway, I'm pretty sure they're only needed for parallel I/O, so can be set UNUSED in the default implementation and overridden by H5Format, or by PncFormat when that is made to work, if it is indeed a bug that they are missing in PncFormat.

Reply:
Ok, I'm happy that this isn't a bug and `UNUSED` is the correct thing to do here